### PR TITLE
reworked drills

### PR DESCRIPTION
**Pull request by J3fftw1**
## Short Description

reworked the drills and combined the events

## Additions/Changes/Removals

combined the drill events

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ofcourse not its me] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 1]
```diff
@@ -161,101 +158,98 @@ public void onDye(PlayerInteractEntityEvent e) {
...
     @EventHandler
-    public void onMiningDrillUse(PlayerInteractEvent e) {
-        final Block block = e.getClickedBlock();
+    public void onDrillUse(PlayerInteractEvent event) {
+        final Block block = event.getClickedBlock();
         if (block == null) {
             return;
         }
 
+        System.out.println("hoi");
```
Comment (Sefiraat) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: Nope

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 2]
```diff
@@ -161,101 +158,98 @@ public void onDye(PlayerInteractEntityEvent e) {
...
+
+
         final Material blockType = block.getType();
         final Location blockLocation = block.getLocation();
+        final ItemStack hand = event.getItem();
+        System.out.println(hand);
```
Comment (Sefiraat) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: Nope

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 3]
```diff
@@ -161,101 +158,98 @@ public void onDye(PlayerInteractEntityEvent e) {
...
-        Validate.notNull(diamondDrill, "Can not be null");
-        if (diamondDrill.isItem(e.getItem())
-            && !diamondDrill.isDisabled()
-            && SlimefunTag.MINEABLE_PICKAXE.isTagged(blockType)
-            && Slimefun.getProtectionManager().hasPermission(e.getPlayer(),
-            blockLocation, Interaction.BREAK_BLOCK)
-        ) {
-            e.setCancelled(true);
+        BlockBreakEvent newEvent = new BlockBreakEvent(block, event.getPlayer());
+        Bukkit.getServer().getPluginManager().callEvent(newEvent);
```
Comment (Sefiraat) [inline src/main/java/dev/j3fftw/litexpansion/Events.java:232]: I know you weren't before, but shouldn't this be checking to see if the event was cancelled?

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 4]
```diff
@@ -161,101 +158,98 @@ public void onDye(PlayerInteractEntityEvent e) {
...
+        block.setType(Material.AIR);
+        event.getPlayer().playSound(blockLocation, Sound.BLOCK_STONE_BREAK, 1.5F, 1F);
 
-            final SlimefunItem slimefunItem = BlockStorage.check(block);
+        // This has to be done because the item in the main hand is not a pickaxe
+        if (blockType == Material.STONE) {
+            blockLocation.getWorld().dropItem(blockLocation,
```
Comment (Sefiraat) [inline src/main/java/dev/j3fftw/litexpansion/Events.java:243]: Need to do something similar for Deepslate and Cobbled Deepslate I'm guessing?

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 5]
```diff
@@ -161,101 +158,95 @@ public void onDye(PlayerInteractEntityEvent e) {
...
-                    blockLocation.getWorld().dropItem(blockLocation,
-                        new ItemStack(blockType)
-                    );
-                }
+            Validate.notNull(diamondDrill);
+            if (!Check(diamondDrill, event, blockLocation)) {
+                return;
+            }
+
+            if (!SlimefunTag.MINEABLE_PICKAXE.isTagged(blockType)) {
```
Comment (Sefiraat) [inline src/main/java/dev/j3fftw/litexpansion/Events.java:190]: Not sure here, but this looks redundant as you cant get here without being included in STONE_VARIANTS.

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 6]
```diff
@@ -161,101 +158,95 @@ public void onDye(PlayerInteractEntityEvent e) {
...
         if (block == null) {
             return;
         }
 
+
         final Material blockType = block.getType();
         final Location blockLocation = block.getLocation();
+        final ItemStack hand = event.getItem();
+
+        if (!SlimefunTag.STONE_VARIANTS.isTagged(blockType)) {
```
Comment (Sefiraat) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: STONE_VARIANTS do not inclide any of the newer stone-style blocks which, likely shouldn't be tackled here, but you may wanna PR to make this tool better? Thats just IMO though, I guess.

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: mmmh so pr to improve STONE_VARIANTS or add deepslate variants?

Comment (Sefiraat) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: I mean, improving STONE_VARIANTS would benefit more, imo :)

File: src/main/java/dev/j3fftw/litexpansion/Events.java (hunk 5 above)
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java:190]: MINEABLE_PICKAXE includes all blocks that the pickaxe can break having it redundant wont actually matter to much but i will move it to inside the miningDrill

File: src/main/java/dev/j3fftw/litexpansion/Events.java (hunk 6 above)
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: will do

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 7]
```diff
@@ -161,101 +157,101 @@ public void onDye(PlayerInteractEntityEvent e) {
...
+            if (!SlimefunTag.MINEABLE_PICKAXE.isTagged(blockType)) {
+                return;
+            }
+
+            DrillUse(1.5f, block, blockType, blockLocation, event);
         }
     }
+    
+    public boolean Check(MiningDrill miningDrill, PlayerInteractEvent event, Location blockLocation) {
+        return miningDrill.isItem(event.getItem())
```
Comment (Sefiraat) [inline src/main/java/dev/j3fftw/litexpansion/Events.java:200]: So, I may be wrong here - but you have covered isItem() before you get here, hasPermission() is covered by the event.isCancelled (I think, check this!). So I think (think!) this method can be replaced with an in-line isDisabled (which if you moved this whole thing into an ItemHandler you wouldnt't even need that - but thats a whoooole other conversation.

Sorry if im wrong on this - im really just glancing inbetween cooking dinner bits xD

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java:200]: i tested and this isnt the case

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 8]
```diff
@@ -161,101 +156,102 @@ public void onDye(PlayerInteractEntityEvent e) {
...
             return;
         }
 
         final Material blockType = block.getType();
         final Location blockLocation = block.getLocation();
+        final ItemStack hand = event.getItem();
 
         final MiningDrill miningDrill = (MiningDrill) SlimefunItem.getById(Items.MINING_DRILL.getItemId());
+        if (miningDrill.isItem(hand)) {
+            Validate.notNull(miningDrill);
```
Comment (TheBusyBiscuit) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: This validate check won't fire, if `miningDrill` is null, then the if-statement will already trigger a NPE.
Either place the validate check before this, use `Objects.notNull` inside the if-statement or remove the validate check.

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 9]
```diff
@@ -161,101 +156,102 @@ public void onDye(PlayerInteractEntityEvent e) {
...
         }
     }
 
-    //TODO combine onDiamondDrillUse with onMiningDrillUse
-    @EventHandler
-    public void onDiamondDrillUse(PlayerInteractEvent e) {
-        final Block block = e.getClickedBlock();
-        if (block == null) {
+
+    public boolean Check(MiningDrill miningDrill, PlayerInteractEvent event, Location blockLocation) {
```
Comment (TheBusyBiscuit) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: Method naming conventions :eyes:

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 10]
```diff
@@ -161,101 +156,102 @@ public void onDye(PlayerInteractEntityEvent e) {
...
+        return miningDrill.isItem(event.getItem())
+            && !miningDrill.isDisabled()
+            && Slimefun.getProtectionManager().hasPermission(event.getPlayer(),
+            blockLocation, Interaction.BREAK_BLOCK);
+    }
+
+    public void DrillUse(float charge, Block block, Material blockType,
```
Comment (TheBusyBiscuit) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: Method naming conventions :eyes:

**Review by TheBusyBiscuit — DISMISSED** [review]

File: src/main/java/dev/j3fftw/litexpansion/Events.java (hunk 10 above)
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: Same as above

File: src/main/java/dev/j3fftw/litexpansion/Events.java (hunk 9 above)
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: whoops, i didnt see this when my friend wrote this code.

Again Thanks for the feedback <3

File: src/main/java/dev/j3fftw/litexpansion/Events.java (hunk 8 above)
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: Thanks for the feedback <3

File: src/main/java/dev/j3fftw/litexpansion/Events.java [hunk 11]
```diff
@@ -168,43 +168,43 @@ public void onDrillUse(PlayerInteractEvent event) {
...
             }
 
-            DrillUse(0.5f, block, blockType, blockLocation, event);
+            drillUse(0.5f, block, blockType, blockLocation, event);
         }
 
         final MiningDrill diamondDrill = (MiningDrill) SlimefunItem.getById(Items.DIAMOND_DRILL.getItemId());
         if (diamondDrill.isItem(hand)) {
 
             Validate.notNull(diamondDrill);
```
Comment (TheBusyBiscuit) [inline src/main/java/dev/j3fftw/litexpansion/Events.java]: Same with this `Validate.notNull` btw

Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: i am not gonna repeat my self :D

File: src/main/java/dev/j3fftw/litexpansion/Events.java (hunk 8 above)
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/Events.java]: Thanks for the feedback

**Comment by J3fftw1** [comment]
would you say lgtm @TheBusyBiscuit

**Pull request merged**